src/Console/MergeContacts.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Console;
+
+use Friendica\Core\L10n;
+use Friendica\Database\Database;
+
+/**
+ * tool to find and merge duplicated contact entries.
+ */
+class MergeContacts extends \Asika\SimpleConsole\Console
+{
+	protected $helpOptions = ['h', 'help', '?'];
+
+	/**
+	 * @var $dba Database
+	 */
+	private $dba;
+
+	/**
+	 * @var L10n
+	 */
+	private $l10n;
+
+	protected function getHelp()
+	{
+		$help = <<<HELP
+console mergecontacts - Merge duplicated contact entries
+Synopsis
+	bin/console mergecontacts
+
+Description
+	bin/console mergecontacts
+		Remove duplicated contact entries
+
+Options
+	-h|--help|-? Show help information
+	-e|--execute Execute the dropping
+HELP;
+		return $help;
+	}
+
+	public function __construct(Database $dba, L10n $l10n, array $argv = null)
+	{
+		parent::__construct($argv);
+
+		$this->dba  = $dba;
+		$this->l10n = $l10n;
+	}
+
+	protected function doExecute()
+	{
+		$duplicates = $this->dba->p("SELECT COUNT(*) AS `total`, `uri-id`, MAX(`url`) AS `url` FROM `contact` WHERE `uid` = 0 GROUP BY `uri-id` HAVING total > 1");
```

Comment:
If we don't need to retrieve the actual total we don't need the COUNT in the SELECT.
```suggestion
		$duplicates = $this->dba->p("SELECT `uri-id`, MAX(`url`) AS `url` FROM `contact` WHERE `uid` = 0 GROUP BY `uri-id` HAVING COUNT(*) > 1");
```

Reply:
It is used in the output